js/ui/windowManager.js

```diff
@@ -2948,6 +2948,20 @@ var WindowManager = new Lang.Class({
             return;
         }
 
+        if (actor._animatableSurface) {
+            // This is an event that may cause an animation
+            // on the window.
+            let attachedEffect = actor._animatableSurface.highest_priority_attached_effect_for_event('close');
+
+            if (attachedEffect) {
+                attachedEffect.activate('close', {
+                    pointer: global.get_pointer(),
+                    complete: () => shellwm.completed_destroy(actor)
+                });
```

Comment:
Same comment about a helper function (I know maybe now we need to have optional args because this one uses `pointer`, not `icon`).

Reply:
Done